ocs_ci/ocs/resources/test_packagemanifest.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf8 -*-
+
+import pytest
+
+from ocs_ci.ocs.resources.packagemanifest import PackageManifest
+from ocs_ci.ocs.exceptions import ResourceNameNotSpecifiedException
+
+
+def test_pm_null():
+    pm = PackageManifest()
+    assert pm is not None
+
+
+def test_pm_null_get_default_channel():
+    pm = PackageManifest()
+    with pytest.raises(ResourceNameNotSpecifiedException):
+        pm.get_default_channel()
+
+
+def test_pm_get_default_channel_missing():
+    """
+    Test that when we run into issue #1338, PackageManifest object still
+    behaves in the same way as before.
+
+    This unit test serves two purposes:
+    - to show what exactly happens to PackageManifest during issue #1338
+    - demonstrate that PackageManifest API remains unchanged
+    """
+    pm = PackageManifest(resource_name='foobar')
+    # based on value of _data attribute when packagemanifest data are missing
+    # as reported in https://github.com/red-hat-storage/ocs-ci/issues/1338
+    pm._data = {
```

Comment:
I think that to this situation you can get only when you have the selector specified as well. In case you specify only resource_name and there won't be found one you will fail most likely directly with commandFailed exception as it will fail when `oc get packagemanifest resource_name`. When you use selector you can expect more results.

I think that in the case we have specified the selector + resource name we should probably add one more condition here:
https://github.com/red-hat-storage/ocs-ci/pull/1475/files#diff-86c9c069ed728ccd16d3f856db567f5eR47
like 
```python
            selector = kwargs.get('selector')
            if data_len == 0 and selector and resource_name:
                raise ResourceNotFound(
                    f"Packagemanifest {resource} with selector {selector} "
                    f"not found!"
                )
```

This case we will know that the pacakge manifest is missing because if we specify selector and name we probably expect at least one result.  WDYT?

This case we will fail in first call of get with that exception and we do not have to deal with any other key error as if we do not load any resource we shouldn't even look for specific data.

Reply:
In this test case, I'm concerned with use case which happens with PackageManifest object when I hit issue #1338 only. For this reason, I set data attribute accordingly and check what happens when `get_default_channel` is called.

Reply:
Also in this pull request, I'm interested only in error handling in ocs-ci pytest plugin and in get_(default)_channel(s) methods. If you want to fix the underlying problem, let's to that in a different PR.

Reply:
If we will do fix like I suggested we will never get to this situation as we will have the exception that resource doesn't exist which will make more sense to me and informs user more than any other .

To make the proper test for this I think that you will have to mock the get function from OCP object to return this data and then tests how the rest of functionality from overloaded get method from PackageManifest.

I am still for the fix I suggested  above:
```python
            selector = kwargs.get('selector')
            if data_len == 0 and selector and resource_name:
                raise ResourceNotFound(
                    f"Packagemanifest {resource} with selector {selector} "
                    f"not found!"
                )
```

Will you cover here or should I send it in different PR?

Reply:
> If we will do fix like I suggested we will never get to this situation as we will have the exception that resource doesn't exist which will make more sense to me and informs user more than any other .

The fix you describe makes sense, but it is not a good reason to not handle problems with get channel methods in pytest plugin code.

> Will you cover here or should I send it in different PR?

I suggest to merge this to unblock test runs on vmware, and you can send another PR with the fix you describe.